apps/api/src/resources/Event/resolver.ts

```diff
@@ -152,9 +152,33 @@ export class EventResolver {
       newResource.flierLink = url;
     }
 
+    let charPool: string = "0123456789abcdefghijklmnopqrstuvwxyz";
+    let keyPlaceholder: string = "";
+    for(let i = 0; i < 4; i++)
+      keyPlaceholder += charPool.charAt(Math.floor(Math.random() * 36));
+    newResource.urlKey = keyPlaceholder;
+
     return newResource.save();
   }
 
+  @Authorized()
+  @Mutation((_: void) => Event)
+  public async addAttendee(
+    @Arg("userId") userId: string,
+    @Arg("eventId") eventId: number,
+  ): Promise<Event> {
+    const event: Event = await Event.findOneOrFail({ id: eventId });
+    let users: User[] = await event.attendees;
+
+    for(let i = 0; i < users.length; i++)
+      if(users[i].id == userId)
+        return event;
+
+    users.push(await User.findOneOrFail({ id: userId }));
```

Comment:
this is fine if you want this function to be used by admins of an event, but if the users themselves are doing the attendance, then you will want to make another function that takes the user from the context (see line 116) and adds their attendance. You will want to add a permission string to the authorized call for this function as well to keep it secure for admins